pymatgen/io/vasp/inputs.py

```diff
@@ -419,16 +420,36 @@ def from_str(cls, data, default_names=None, read_velocities=True):
                 for i, nat in enumerate(n_atoms):
                     sym = Element.from_Z(i + 1).symbol
                     atomic_symbols.extend([sym] * nat)
-                warnings.warn(f"Elements in POSCAR cannot be determined. Defaulting to false names {atomic_symbols}.")
-        # read the atomic coordinates
+                warnings.warn(
+                    f"Elements in POSCAR cannot be determined. Defaulting to false names {atomic_symbols}.",
+                    BadPoscarWarning,
+                )
+
+        # Read the atomic coordinates
         coords = []
         selective_dynamics = [] if has_selective_dynamics else None
         for i in range(n_sites):
             tokens = lines[ipos + 1 + i].split()
             crd_scale = scale if cart else 1
             coords.append([float(j) * crd_scale for j in tokens[:3]])
             if has_selective_dynamics:
-                selective_dynamics.append([tok.upper()[0] == "T" for tok in tokens[3:6]])
+                # Warn when values contain suspicious entries
+                if any(value not in {"T", "F"} for value in tokens[3:6]):
+                    warnings.warn("Selective dynamics values must be either 'T' or 'F'.", BadPoscarWarning)
+
+                # Warn when elements contains Fluorine (F) (#3539)
+                if atomic_symbols[i] == "F" and len(tokens[3:]) >= 4 and "F" in tokens[3:7]:
+                    warnings.warn(
+                        "Selective dynamics with Fluorine. Make sure the 4th-6th entry is selective dynamics.",
+                        BadPoscarWarning,
+                    )
+
+                selective_dynamics.append([value == "T" for value in tokens[3:6]])
+
+        # Warn when ALL degrees of freedom relaxed (#3539)
+        if has_selective_dynamics and all(all(i is True for i in in_list) for in_list in selective_dynamics):
+            warnings.warn("Selective dynamics toggled with ALL degrees of freedom relaxed.", BadPoscarWarning)
```

Comment:
Maybe make this warning more descriptive: "Selective dynamics can only be used when at least one degree of freedom is not permitted to relax. All degrees of freedom are currently allowed to relax."

Reply:
Hi @esoteric-ephemera thanks for reviewing and the suggestion. Would that be a little too verbose? I think we just need to make users aware that they might be doing something improper? Also to avoid confusion from the user such as in the original issue #3008?


Meanwhile I think `VASP` would be perfectly happy to toggle selective dynamics with all DOFs relaxed, so I don't quite agree with the following message:

> Selective dynamics can only be used when at least one degree of freedom is not permitted to relax.

Also from [VASP wiki](https://www.vasp.at/wiki/index.php/POSCAR):

> If the selective dynamics feature is enabled on each coordinate triplet is followed by three additional logical flags, i.e. each is either T or F for true and false, respectively. This determines whether to allow changes of the coordinates or not. If the line selective dynamics is removed from the POSCAR file this flag will be ignored (and internally set to T).

So I think what we should do is to remain the user some change (in this case the selective dynamics flag would be dropped) is expected to avoid confusion. What do you think?

Reply:
To avoid confusion, I think it's better to have a descriptive warning message that explains why the warning is thrown. "Ignoring `selective dynamics` tag, as no ionic degrees of freedom were disabled." is maybe clearer and more to the point.

Just a suggestion, hope you're feeling better!